src/cdk-experimental/drag-drop/drag.ts

```diff
@@ -0,0 +1,493 @@
+    }
+
+  /**
+   * Returns the element that is being used as a placeholder
+   * while the current element is being dragged.
+   */
+  getPlaceholderElement(): HTMLElement {
+    return this._placeholder;
+  }
+
+  ngAfterContentInit() {
+    // TODO: doesn't handle (pun intended) the handle being destroyed
+    const element = (this._handle ? this._handle.element : this.element).nativeElement;
+    element.addEventListener('mousedown', this._pointerDown);
+    element.addEventListener('touchstart', this._pointerDown);
+
+    // Webkit won't preventDefault on a dynamically-added `touchmove` listener, which means that
+    // we need to add one ahead of time. See https://bugs.webkit.org/show_bug.cgi?id=184250.
+    // TODO: move into a central registry.
+    this._ngZone.runOutsideAngular(() => {
+      this._document.addEventListener('touchmove', this._preventScrollListener, activeEventOptions);
+    });
+  }
+
+  ngOnDestroy() {
+    this._removeDocumentEvents();
+    this._destroyPreview();
+    this._destroyPlaceholder();
+    this._document.removeEventListener('touchmove', this._preventScrollListener,
+        activeEventOptions as any);
+
+    if (this._isDragging) {
+      // Since we move out the element to the end of the body while it's being
+      // dragged, we have to make sure that it's removed if it gets destroyed.
+      this._removeElement(this.element.nativeElement);
+    }
+  }
+
+  /** Handler for when the pointer is pressed down on the element or the handle. */
+  private _pointerDown = (event: MouseEvent | TouchEvent) => {
+    if (this._isDragging) {
+      return;
+    }
+
+    this._isDragging = true;
+    this._initialContainer = this.dropContainer;
+
+    // If we have a custom preview template, the element won't be visible anyway so we avoid the
+    // extra `getBoundingClientRect` calls and just move the preview next to the cursor.
+    this._pickupPositionInElement = this._previewTemplate ? {x: 0, y: 0} :
+        this._getPointerPositionInElement(event);
+    this._pickupPositionOnPage = this._getPointerPositionOnPage(event);
+
+    this._ngZone.runOutsideAngular(() => {
+      const isTouchEvent = this._isTouchEvent(event);
+
+      // We explicitly bind active listeners here, because newer browsers
+      // will default to passive ones for `mousemove` and `touchmove`.
+      // TODO: this should be bound in `mousemove` and after a certain threshold,
+      // otherwise it'll interfere with clicks on the element.
+      this._document.addEventListener(isTouchEvent ? 'touchmove' : 'mousemove', this._pointerMove,
+          activeEventOptions);
+      this._document.addEventListener(isTouchEvent ? 'touchend' : 'mouseup', this._pointerUp);
+    });
+
+    // Emit the event on the item before the one on the container.
+    this.started.emit({source: this});
+
+    if (this.dropContainer) {
+      const element = this.element.nativeElement;
+      const preview = this._preview = this._createPreviewElement();
```

Comment:
Rather than the multiple assignment here, should `_createPreviewElement` also assign its output to `this._preview`?

It could be thought of as `setupPreviewElement` rather than just creating.  I think this especially makes sense since `_createPreviewElement` already has side effects of creating the refs needed for the preview.

Reply:
I guess this is more of a personal preference. I prefer not to have functions initialize properties like this since it isn't very transparent.